### PR TITLE
[openid][stable-3_4_0] #1 Initial Commit - port to stable-3_4_0

### PR DESCRIPTION
**Pull request by defstat**

### CODE REVIEW

**Review by asmecher — CHANGES_REQUESTED** [review]
Just a few comments, thanks!

File: OpenIDPlugin.php [hunk 1]
```diff
@@ -0,0 +1,364 @@
+<?php
+
+/**
+ * @file OpenIDPlugin.php
+ *
+ * Copyright (c) 2020 Leibniz Institute for Psychology Information (https://leibniz-psychology.org/)
```
[the remark this answers is not in the document]
Comment (defstat) [reply OpenIDPlugin.php:6]: For existing files I updated the copyright to 

```
 * Copyright (c) 2020 Leibniz Institute for Psychology Information (https://leibniz-psychology.org/)
 * Copyright (c) 2024 Simon Fraser University
 * Copyright (c) 2024 John Willinsky
 * Distributed under the GNU GPL v3. For full terms see the file LICENSE.
```

For new files (like `classes/ContextData.php`) I added the copyright:

```
 * Copyright (c) 2024 Simon Fraser University
 * Copyright (c) 2024 John Willinsky
 * Distributed under the GNU GPL v3. For full terms see the file LICENSE.
```

File: OpenIDPlugin.php [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+ * Copyright (c) 2023 Simon Fraser University
+ * Copyright (c) 2023 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file LICENSE.
+ *
+ * @class OpenIDPlugin
+ *
+ * @ingroup plugins_generic_openid
```
[the remark this answers is not in the document]
Comment (defstat) [reply OpenIDPlugin.php:13]: Removed it from all files

File: handler/OpenIDHandler.php [hunk 3]
```diff
@@ -0,0 +1,499 @@
...
+
+		if ($errorMsg) {
+			$ssoErrors['sso_error_msg'] = $errorMsg;
+		}
+
+
+		return $request->redirect($contextPath, 'login', null, null, $ssoErrors);
+	}
+
+	public static function getOpenIDSettings(OpenIDPlugin $plugin, int $contextId): ?array
```
[the remark this answers is not in the document]
Comment (defstat) [reply handler/OpenIDHandler.php:486]: Moved to `OpenIDPlugin` and updated the referencing code

**Pull request merged**